plugins/discovery-ec2/src/main/java/org/opensearch/discovery/ec2/AwsEc2ServiceImpl.java

```diff
@@ -121,26 +173,36 @@ public AmazonEc2Reference client() {
      */
     @Override
     public void refreshAndClearCache(Ec2ClientSettings clientSettings) {
-        final LazyInitializable<AmazonEc2Reference, OpenSearchException> newClient = new LazyInitializable<>(
-            () -> new AmazonEc2Reference(buildClient(clientSettings)),
+        final LazyInitializable<AmazonEc2ClientReference, OpenSearchException> newClient = new LazyInitializable<>(
+            () -> new AmazonEc2ClientReference(buildClient(clientSettings)),
             clientReference -> clientReference.incRef(),
             clientReference -> clientReference.decRef()
         );
-        final LazyInitializable<AmazonEc2Reference, OpenSearchException> oldClient = this.lazyClientReference.getAndSet(newClient);
+        final LazyInitializable<AmazonEc2ClientReference, OpenSearchException> oldClient = this.lazyClientReference.getAndSet(newClient);
         if (oldClient != null) {
             oldClient.reset();
         }
     }
 
     @Override
     public void close() {
-        final LazyInitializable<AmazonEc2Reference, OpenSearchException> clientReference = this.lazyClientReference.getAndSet(null);
+        final LazyInitializable<AmazonEc2ClientReference, OpenSearchException> clientReference = this.lazyClientReference.getAndSet(null);
         if (clientReference != null) {
+            clientReference.getOrCompute().get().close();
```

Comment:
It think using    `clientReference.getOrCompute()` is not needed here:
 - `clientReference.reset();` calls `decRef()`
 - `decRef()` calls `closeInternal();` (when refCount == 0)
 - `closeInternal()` calls `close()`

Reply:
Yes, you're right. Removed.